stl/CMakeLists.txt

```diff
@@ -405,6 +405,11 @@ set(STATIC_SOURCES
     ${SOURCES_SATELLITE_CODECVT_IDS}
 )
 
+# Objs that exist in all satellite DLLs
+set(SATELLITE_DLL_SOURCES

```

Comment:
"DLL" seems redundant to me in `SATELLITE_DLL_SOURCES` and `msvcp{$D_SUFFIX}_satellite_dll_objects`. The fact that all satellites are DLLs doesn't need repetition.

Reply:
I think it's potentially useful for readers who don't know a lot about our DLL schemes. Unlike product code, where we can assume a high level of familiarity with the Standard and can therefore be very terse sometimes, things like the build system should probably err on the side of being too descriptive and self-documenting. If we never said "DLL" next to "satellite", readers wouldn't necessarily know what they are.

Reply:
I don't mind the "DLL" in the comment, which I think makes that connection for readers. What concerns me is that we have lots of names that include "satellite" without "dll", and now two occurrences of "satellite_dll". The anomaly suggests there's something deeper going on. (I've now expended a satisfactory amount of effort on this hill.)

Reply:
There is actually supposed to be a distinction here; for example `${SOURCES_SATELLITE_1}` is shared with both the satellite DLL, and also with the static lib for the static library versions, whereas anything marked `_dll` is supposed to be specific to the DLL only. I observe that some places are missing that, will fix.

Reply:
Although I guess this whole function is about DLLs. I'll remove it from the target name but I think the variable name should keep it for the reason above.

Reply:
> Unlike product code, where we can assume a high level of familiarity with the Standard

😄 
I guess most of people who are staring at STL code are debugging their own code or analyzing their crash dump.
Think it happens more often and for longer time with inexperienced programmers.

I don't object quoting Standard, at it can always be reached; but using some inner knowledge and using too terse language may not be helpful.

Reply:
(Guess how many times the question #1006 was independently asked, for example)